include/seqan3/alignment/matrix/alignment_trace_algorithms.hpp

```diff
@@ -119,8 +116,18 @@ alignment_trace(database_t && database,
     assert(row < matrix.rows());
     assert(col < matrix.cols());
 
-    std::deque<gapped_database_alphabet_t> gapped_database{};
-    std::deque<gapped_query_alphabet_t> gapped_query{};
+    auto begin_db = std::ranges::begin(database);
+    auto end_db = std::ranges::end(database);
+    auto begin_query = std::ranges::begin(query);
+    auto end_query = std::ranges::end(query);
+
+    alignment_t aligned_seq;
+    assign_unaligned(std::get<0>(aligned_seq),
+                     std::ranges::subrange<decltype(begin_db), decltype(end_db)>(begin_db, end_db));
+    assign_unaligned(std::get<1>(aligned_seq),
+                     std::ranges::subrange<decltype(begin_query), decltype(end_query)>(begin_query, end_query));
+    auto it_db = cend(std::get<0>(aligned_seq));
```

Comment:
```suggestion
    auto end_db = std::ranges::cend(std::get<0>(aligned_seq));
```
? to be consistent to begin_db.

Reply:
Why cend and not end`

Reply:
instead of calling std::ranges::end here, call std::ranges::begin + std::ranges::size so that you know you can go backwards.

Reply:
> Why cend and not end`

insert_gap of [AlignedSequence](http://docs.seqan.de/seqan/3-master-dev/interfaceseqan3_1_1AlignedSequence.html) requires a const_iterator, so I have to use `cbegin` and `cend`.

Reply:
Oh, and `gap_decorator_iterator` doesn't allow operator+
So I implement a loop that increments size times?

Reply:
> insert_gap of AlignedSequence requires a const_iterator, so I have to use cbegin and cend.

there should be implicit non-cont → const conversion but maybe that's broken

Reply:
> Oh, and gap_decorator_iterator doesn't allow operator+

oh no.

> So I implement a loop that increments size times?

no, better not. We know the gap_decorator is always a CommonRange so just stick with end() and decrement that.